manifests/init.pp

```diff
@@ -131,30 +132,36 @@
     # In this code-path, $package_version != undef AND we are not on a PE infrastructure
     # node since $::pe_server_version is not defined
 
-    if $package_version !~ /^\d+\.\d+\.\d+([.-]?\d*|\.\d+\.g[0-9a-f]+)$/ {
-      fail("invalid version ${package_version} requested")
+    if $::architecture == 'x86' and $arch == 'x64' {
+      fail('Unable to install x64 on a x86 system')
     }
 
-    # Strip git sha from dev builds
-    if $package_version =~ /g/ {
-      $_expected_package_version = split($package_version, /[.-]g.*/)[0]
+    if $package_version == 'master' {
+      $master_or_package_version = $serverversion
```

Comment:
I don't know where `$serverversion` comes from

Reply:
> $serverversion — the current version of Puppet on the Puppet master.

https://puppet.com/docs/puppet/5.5/lang_facts_and_builtin_vars.html#puppet-master-variables

Like `pe_compiling_server_aio_build` in `params.pp` but it does not require PE for ...

https://github.com/puppetlabs/puppet-enterprise-modules/blob/irving/modules/puppet_enterprise/lib/puppet/parser/functions/pe_compiling_server_aio_build.rb

and

https://github.com/puppetlabs/puppet-enterprise-modules/blob/irving/modules/puppet_enterprise/lib/shared/aio_build.rb

Reply:
Ah, cool. Thank you!

Reply:
I think there was a period in time when puppet version and aio package version weren't guaranteed to match, but I believe they should for recent and future versions.